src/magellan_vision/scripts/line_detector.py

```diff
@@ -1,211 +1,142 @@
-    left_line_y = []
-    right_line_x = []
-    right_line_y = []
-
-    crop = frame
-
-    for line in linesP:
-        for x1, y1, x2, y2 in line:
-            if i == 0:
-                i += 1
-                max_y = y2
-                min_y = y1
-            if y2 < min_y:
-                min_y = y1
-            if y1 < max_y:
-                max_y = y2
-            slope = (y2 - y1) / (x2 - x1)  # <-- Calculating the slope.
-        if math.fabs(slope) < 0.8:  # <-- Only consider extreme slope
-            continue
-        if x1 < frame.shape[1]/2 and x2 < frame.shape[1]/2:  # <-- If the slope is negative, left group.
-            left_line_x.extend([x1, x2])
-            left_line_y.extend([y1, y2])
-        else:  # <-- Otherwise, right group.
-            right_line_x.extend([x1, x2])
-            right_line_y.extend([y1, y2])
-    # TODO: Dynamically assign min and max y, using the iteration above???
-    min_y = (int)(frame.shape[0] * (3/5))
-    max_y = frame.shape[0]
-
-    # TODO: Instead of poly, maybe try cv2.rectangle to match the area covered by linesP more accurately
-    if len(left_line_x) != 0 and len(left_line_y) != 0 and len(right_line_y) != 0 and len(right_line_x) != 0:
-        poly_left = np.poly1d(np.polyfit(
-            left_line_y,
-            left_line_x,
-            deg=1
-        ))
-        left_x_start = int(poly_left(max_y))
-        left_x_end = int(poly_left(min_y))
-
-        poly_right = np.poly1d(np.polyfit(
-            right_line_y,
-            right_line_x,
-            deg=1
-        ))
-        right_x_start = int(poly_right(max_y))
-        right_x_end = int(poly_right(min_y))
-        # TODO: Filter out "bad" lines (i.e. if the two lines intersect)
-        crop = draw_lines(
-            frame,
-            [[
-                [left_x_start, max_y, left_x_end, min_y],
-                [right_x_start, max_y, right_x_end, min_y],
-            ]],
-            thickness=5,
+class Obstacle:
+
+    def __init__(self, type_of_obstacle, color_of_obstacle, x_center, y_center):
+        self.type_of_obstacle = type_of_obstacle
+        self.color_of_obstacle = color_of_obstacle
+        self.x_center = x_center
+        self.y_center = y_center
+        self.dist = -1  # default to -1 so we can check later if the distance has been verified
+
+
+class Lines:
+
+    def __init__(self):
+        self.centers = 0
+
+    def skeletize(self, img, size, skel):
+        element = cv2.getStructuringElement(cv2.MORPH_CROSS, (3, 3))
+        done = False
+        while(not done):
+            eroded = cv2.erode(img, element)
+            temp = cv2.dilate(eroded, element)
+            temp = cv2.subtract(img, temp)
+            skel = cv2.bitwise_or(skel, temp)
+            img = eroded.copy()
+            zeros = size - cv2.countNonZero(img)
+            if zeros == size:
+                done = True
+        return skel
+
+    def draw_lines(img, lines, color=[0, 255, 0], thickness=3):
```

Comment:
Same as above for private variables and methods. Add a `_` at the beginning. Ex. `self._type_of_obstacle`

Reply:
fixed